### PR TITLE
Added Refactoring changes

### PR DESCRIPTION
**Pull request by guryashdhall**
Details of the Refactoring are present in the issue:- https://github.com/OSSpk/Library-Management-System-JAVA/issues/9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ The **Class Diagram** of the project is also provided along with the **Database
 ## Class Diagram
 ![class diagram](../master/images/diagram.PNG)
 
+**Note**: After Refactoring, new Class "HoldRequestOperations" is added to the above structure which lies in between the HoldRequest class and Book class.
+This class removes the bidirectional dependency between HoldRequest and Book. 
+
```
Comment (harismuneer) [inline README.md:20]: @guryashdhall can we add why we made this new class and the context for this decision. Will be helpful for the readers.

**Review by harismuneer — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (guryashdhall) [reply README.md:20]: @harismuneer  sure i will update the same

Comment (guryashdhall) [reply README.md:20]: @harismuneer  I have updated the reasoning behind the introduction of new class in the README

**Comment by harismuneer** [comment]
@guryashdhall I am merging the PR with the note that a subsequent code review from your peers would be required to ensure integrity.

**Comment by amansbhandari** [comment]
@harismuneer  I have reviewed the changes and tested the functionality , the changes made by @guryashdhall are just refactoring changes and has no effect on the working of the Library Management System

**Pull request merged**